### ISSUE TITLE
Data table: Implement inline editing 

### ISSUE THREAD

**Issue by johanna-koval** [carbon-design-system/ibm-cloud-cognitive#1924]
# Data table extension: Inline editing

## Description 

WKC data table does not include any of the features and behaviors of the PAL Inline editing pattern:
https://pages.github.ibm.com/cdai-design/pal/components/data-table/inline-editing/usage

There is an initial prototype of some of the edit behaviors in this storybook:
https://pages.github.ibm.com/Jan-Hassel/spss-prototype-components/?path=/story/components-variableview--default

Additional details from #1616 

> ## Details
> [Link to Design](https://pages.github.ibm.com/cdai-design/pal/components/data-table/inline-editing)
> 
> ## Design Maintainer(s)
> * Marion Brülls
> * Virginia Nicholson
> * Mitchell Bernstein
> 
> ## Tasks
> Before starting work on this epic, please review and complete the following.
> 
> * [ ]  Initial review of design/existing code
> 
> ### Working in Carbon for IBM Products package
> * [ ]  Have you reviewed our
>   [contribution guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/.github/CONTRIBUTING.md)?
> * [ ]  Have you noted our
>   [code guidelines](https://github.com/carbon-design-system/ibm-cloud-cognitive/blob/master/docs/CODE_GUIDELINES.md)?
> * [ ]  Did you use the CLI to scaffold your new component?

**Comment by David-Yuen** [comment]
So i've manually implemented using the InlineEdit, found quite a few issues and I hope you guys resolve them. Once your implementation here is ready, I hope I can just replace my current InlineEdit component with this

**Comment by janhassel** [comment]
@matthewgallo design review based on PR #2269 deploy preview:

| # | Screenshot | Description |
| :- | :- | :- |
| 1 | ![Screen Recording 2022-09-15 at 09 40 11](https://user-images.githubusercontent.com/28265588/190345215-a5cb6dd7-631c-4be2-add5-910f5d23efcf.gif) | The region focus indicator (lightblue) influences the grid's dimensions |
| 2 | <img width="201" alt="image" src="https://user-images.githubusercontent.com/28265588/190345574-4e035205-5bcb-4a5f-bac6-93aaad139677.png"> | The cursor changed to `pointer` even for cells that don't support editing. Should be `default` |
| 3 | <img width="371" alt="image" src="https://user-images.githubusercontent.com/28265588/190345930-cd0c77bd-56b6-4b79-bdd6-7e781e32a6cb.png"> <img width="193" alt="image" src="https://user-images.githubusercontent.com/28265588/190346005-450e0e4d-e695-4f8c-aacd-2d0dd4808962.png"> | Overflow controls like date picker and dropdown are truncated near the end of the grid |
| 4 | <img width="741" alt="image" src="https://user-images.githubusercontent.com/28265588/190346182-74ff20aa-1b5d-468c-9743-cab30eca2145.png"> | The row should also receive its hover effect (light gray background) when using keyboard navigation |
| 5 | <img width="585" alt="image" src="https://user-images.githubusercontent.com/28265588/190346783-e97028ac-2b5f-4245-8e3d-288286587a5d.png"> | The row should still have its hover effect (light gray background) when a cell is in editing mode |
| 6 | <img width="212" alt="image" src="https://user-images.githubusercontent.com/28265588/190347054-868332c7-4b99-4484-8b26-cc3d608c3270.png"> | Validation state is not up to spec (/ not implement yet?) |
| 7 | <img width="165" alt="image" src="https://user-images.githubusercontent.com/28265588/190350462-95700af7-cb8f-4bf6-9e87-753b39b8dc7f.png"> | The dropdown icons are baseline-aligned, should be center-aligned |


For furher review, could you please...
- ... add a story (or a control) to see how it looks with the standard behaviour of the icons only shwoing on focus and hover?
- ... add a few disabled cells in the story?
- ... add a control to change the grid's row size / height?

**Comment by matthewgallo** [comment]
Thanks for the review @janhassel! I've fixed numbers 1 and 2 on the list. @Ratheeshrajan is working on 4-7. Number 3 is a bit more complex and looks like Carbon has some issues open around similar issues of dropdowns/floating menus inside of scrolling containers. For example, [here](https://github.com/carbon-design-system/carbon/issues/7565).

In terms of the stories, we're in the midst of cleaning up the Datagrid stories this week and next week, so I will make sure that your suggestions are added for the inline editing story/stories.

**Comment by Ratheeshrajan** [comment]
@matthewgallo fixed issues 4,5 and 7 in the following [PR](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/2282)

**Comment by elycheea** [comment]
Will open a separate issue to address 3 from Jan since this affects some other extensions such as filtering as well.

**Comment by matthewgallo** [comment]
Datagrid inline editing support has been added.

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-datagrid-datagrid-canary-extensions-inlineedit--inline-edit-usage-story
